docs/content/v1/usage/lifecycle.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Lifecycle callbacks
+weight: 8
+---
+
+LEGO comes with 5 methods that allows you to hook some callbacks within the component lifecycle.
+
+```js
+export default class extends Lego {
+  init() { … }
+
+  connected() { … }
+
+  rendered() { … }
+
+  changed() { … }
+
+  disconnected() { … }
+}
+```
+
+## `init`
+
+This callback is called on the first time when the element is added to the DOM. At this point the element is still not rendered, so you can't access the [Shadow DOM]((https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM)) and inner elements.
+
+## `connected`
+
+This callback is called when the element is added to the DOM **and** rendered.
+
+## `rendered`
+
+This callback is called **after** each time the element is rendered. It means at the initialization (after `init` and before `connected`), as well as each time the state is updated (with `this.render()`) and the component is re-rendered.
+
+The method is called with one param corresponding to the updated state.
+
+## `changed`
+
+This callback is called when `state` is just mutated, **before** (re-)rendering.
+
+The method is called with one param corresponding to the updated state.
+
+## `disconnected`
+
+This callback is called when the element is removed from the DOM.
+
+## You Said Native Web Components? Let's Hack!
```

Comment:
I am not sure if it is a good to add this section? I just propose it.

Reply:
This callback is triggered when the element is removed from the DOM.